From e9adac4468e3abd0d26bcbd51f80250904f1e685 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 26 Oct 2018 14:09:01 +0200
Subject: [PATCH] arm64: dt: fsl-imx8qxp-tqma8qx: limit cma reserved size to
 256 MiB

needed since we have normally only 1 GiB of RAM

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi    | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
index a3d61179112d..bf29c3c0a14a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx.dtsi
@@ -22,6 +22,20 @@
 	model = "TQ Systems i.MX8QXP TQMa8xX";
 	compatible = "fsl,tqma8qx", "fsl,imx8qxp";
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0 0x10000000>;
+			alloc-ranges = <0 0x96000000 0 0x20000000>;
+			linux,cma-default;
+		};
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
